Composer/packages/client/src/pages/design/DebugPanel/DebugPanel.tsx

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx, css } from '@emotion/core';
+import { useMemo, useCallback, useState } from 'react';
+import { useRecoilState } from 'recoil';
+import formatMessage from 'format-message';
+import { IconButton } from 'office-ui-fabric-react/lib/Button';
+import { Pivot, PivotItem } from 'office-ui-fabric-react/lib/Pivot';
+
+import { debugPanelExpansionState } from '../../../recoilModel';
+
+import {
+  debugPaneContainerExpandedStyle,
+  debugPaneBarStyle,
+  leftBarStyle,
+  rightBarStyle,
+  debugPaneHeaderStyle,
+  statusBarStyle,
+  debugPaneContentStyle,
+} from './styles';
+import debugExtensions from './TabExtensions';
+
+export interface DebugPanelProps {
+  expanded: boolean;
+  onToggleExpansion: (expanded: boolean) => void;
+}
+export const DebugPanel = () => {
+  const [expanded, setExpansion] = useRecoilState(debugPanelExpansionState);
+
+  const [activeTab, setActiveTab] = useState<string>(debugExtensions[0].key);
+
+  const buildTabTitle = useCallback((tabKey: string, TabHeaderWidget: React.FC | string) => {
+    if (!TabHeaderWidget) return { key: tabKey, element: null };
+
+    let element: JSX.Element;
+    if (typeof TabHeaderWidget === 'string') {
+      element = <span key={`tabHeader-${tabKey}`}>{TabHeaderWidget}</span>;
+    } else {
+      element = <TabHeaderWidget key={`tabHeader-${tabKey}`} />;
+    }
+    return { key: tabKey, element };
+  }, []);
+
+  const headerPivot = useMemo(() => {
+    const tabTitles = debugExtensions
+      .map(({ key, HeaderWidget }) => buildTabTitle(key, HeaderWidget))
```

Comment:
nit: Seems like using a forEach to build tabTitles array would replace the two maps and a filter.